viewing_party/party.py

```diff
@@ -0,0 +1,201 @@
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    '''
+    function gets average rating for the movies that are in 
+    users list fo watched movies
+    '''
+    ratings = []
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    else:
+        for movie in user_data["watched"]:
+            ratings.append(movie["rating"])
+        num_of_movies = len(ratings)
+        ratings_total = sum(ratings)
+        avg_rating = ratings_total / num_of_movies
+        return avg_rating
+
+
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) == 0:
+        return None
+    popular_genres = {}
+    for movie in user_data["watched"]:
+        if movie["genre"] not in popular_genres:
+            popular_genres[movie["genre"]] = 1
+        else:
+            popular_genres[movie["genre"]] += 1
+    popularity = 0
+    for genre, popularity_value in popular_genres.items():
+        if popularity_value > popularity:
+            popularity = popularity_value
+            most_popular_genre = genre
+    return most_popular_genre
+
+
+def get_unique_watched(user_data):
+    '''
+    function returns list of movies that are in users watched list
+    but not in any of the users friends watched list. the movies
+    in the returned list are unique to the user.
+    '''
+    friends_watched = get_friends_watched_list(user_data)
+    unique_movies = []
+    for movie in user_data["watched"]:
+        if movie not in friends_watched:
+            unique_movies.append(movie)
+    return unique_movies
+
+
+def get_friends_unique_watched(user_data):
+    '''
+    function returns list of movies that are in user's friends 
+    watched list but not in the users watched list. the movies 
+    in the returned list are unique to the user's friends.
+    '''
+    friends_watched = get_friends_watched_list(user_data)
+    friends_unique_movies = []
+    for movie in friends_watched:
+        if movie not in user_data["watched"] and movie not in friends_unique_movies:
+            friends_unique_movies.append(movie)
+    return friends_unique_movies
+
+
+def get_available_recs(user_data):
+    '''
+    functions gets movie recommendations for user based on the
+    movies that the user's friends have watched. function checks if
+    the movie is in a subscription that the user has and the user 
+    has not watched the movie yet. returns list of recommendations.
+    '''
+    recommendations = []
+    user_subscriptions = user_data["subscriptions"]
+    friends_watched = get_friends_watched_list(user_data)
+    user_watched = []
+    for movie in user_data["watched"]:
+        user_watched.append(movie["title"])
+
+    for movie in friends_watched:
+        if movie["host"] in user_subscriptions and \
+        movie["title"] not in user_watched \
+        and movie not in recommendations:
```

Comment:
I like how you used `\` to split up the compound statements! much easier to read what's going on!